web-app/HeimScripts/heatmap/run.r

```diff
@@ -1,29 +1,90 @@
+  humanReadableNames <-extractNames(datasets)
+  measurements <- assignNames(measurements,humanReadableNames) #combine label with Biomarker(if present) and assign as a name to be displayed on the heatmap
+  measurements <- transform(measurements) #log2 transform the numerical matrix with measurements
+  grouping <- extractGrouping(datasets) #Grouping determines coloring of the samples - one color for dataset1 and a different one for dataset2
+  makeHeatmap(measurements,grouping) # plot the heatmap and save it in the working directory as a .png file
+}
 
-    measurements  <- subset(dataset,select=-c(Row.Label,Bio.marker)) # this will select all columns other than Row.Label,Bio.marker columns
-    measurements  <- data.matrix(measurements)
+parseInput <- function(variables){
+  datasets <- list()
+  varLength <- length(variables)
+  if(varLength > 1){
+    datasets <- list(dataset1 = variables[[1]], dataset2 = variables[[2]])  
+  }
+  else if(varLength == 1){
+    datasets <- list(dataset1 = variables[[1]])
+  }
+  return(datasets)
+}
 
-    measurements <- log(measurements,2)
+extractGrouping <- function(datasets){
+  ds1Length <- ncol(datasets[[1]]) - length(labelColumns)
+  grouping <- rep(dataset1color,ds1Length)
+  if(length(datasets) > 1){
+    ds2Length <- ncol(datasets[[2]]) - length(labelColumns)
+    secondSetGrouping <- rep(dataset2color,ds2Length)
+    grouping <- c(grouping,secondSetGrouping)
+  }
+  return(grouping)
+}
 
-    rownames(measurements) <- df$Row.Label
+extractMeasurements <- function(datasets){
+  measurements <- extractMeasurement(datasets[[1]])
+  if(length(datasets) > 1){
+    secondDsMeasurements <-extractMeasurement(datasets[[2]])
+    measurements <- cbind(measurements,secondDsMeasurements)
+  }
+  return(measurements)
+}
+
+extractMeasurement <- function(dataset){
+  measurements  <- subset(dataset,select=-c(Row.Label,Bio.marker)) # this will select all columns other than Row.Label,Bio.marker columns
+  measurements  <- data.matrix(measurements)
+}
 
-    png(filename="heatmap.png",width = 800,height=800)
-    heatmap.2(measurements,
-          scale = "none",
-          dendrogram = "none",
-          Rowv = NA,
-          Colv = NA,
-          density.info = "none", # histogram", # density.info=c("histogram","density","none")
-          trace = "none",
-          col=redgreen(75),
-          margins=c(12,12)
-          #ColSideColors= as.character(groups)
-          #adjCol=c("left","top")
-    )
-    dev.off()
+assignNames <- function(measurements, humanReadableRowNames){
+  rownames(measurements) <- humanReadableRowNames
+  return(measurements)
 }
 
+extractNames <- function(datasets){
+  humanReadableNames <- datasets[[1]]$Row.Label  
+#  if(length(datasets) > 1){
+#    namesSecondDs <- datasets[[2]]$Row.Label 
+#    humanReadableNames <- c(humanReadableNames, namesSecondDs)  
+#  }
+  return(humanReadableNames)
+}
+
```

Comment:
Can we always assume that the rownnames are the same for two datasets? In case of gene expression data they should be of the same platform, so it should theoretically be the same, unless some filtering on the rows of the data has been done before data upload. Maybe build in a check, just in case?